src/sentry/notifications/notificationcontroller.py

```diff
@@ -58,14 +61,24 @@ def __init__(
         organization_id: int | None = None,
         type: NotificationSettingEnum | None = None,
         provider: ExternalProviderEnum | None = None,
+        organization: Organization | None = None,
     ) -> None:
-        self.recipients = recipients
+        if features.has("organizations:team-workflow-notifications", organization):
+            self.recipients = []
+            for recipient in recipients:
+                if recipient_is_team(recipient):
```

Comment:
@isabellaenriquez where do we fall back to load the members of the team if there isn't a linked slack channel?

Reply:
thanks so much for pointing this out! 🥔  i had assumed that the fallback was handled by another ticket but i've clarified with jodi that it's part of this one. will add that logic today!